RocketChat/Rocket.Chat.ReactNative
# feat: Refactor Status to make it handle user presence requests

## Proposed changes

- Status/index now calls getUserPresence instead of RoomItem
- Added RoomTypeIcon stories
- Added Status stories

## Issue(s)	

Depends on https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/5162

## How to test or reproduce

- Make sure user presence (aka status) is working throughout the app
- This PR also fixes a bug:
  * go to a channel
  * tap on a user avatar you haven't seen yet (so you didn't load its presence)
  * app navigates to user info and fetches their status correctly (previously it would show as offline)

## Screenshots

![image](https://github.com/RocketChat/Rocket.Chat.ReactNative/assets/804994/3ac3b524-6cbf-45c5-b037-a70a49e2196f)
![image](https://github.com/RocketChat/Rocket.Chat.ReactNative/assets/804994/874e01dc-420c-4327-9f64-555955e7aae3)


## Types of changes




## Checklist



## Further comments

[NATIVE-371]

[NATIVE-371]: https://rocketchat.atlassian.net/browse/NATIVE-371?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
